src/components/PeopleList/PeopleList.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, useEffect, useState } from 'react';
+import PeopleItem from '../PeopleItem';
+import { getPeople } from '../../api';
+import { Person } from '../../types/Person';
+
+interface Props {
+  setOnLoad: (state: boolean) => void;
+  setServerError: (state: boolean) => void;
+  setPeopleLength: (state: number) => void;
+}
+
+export const PeopleList: FC<Props> = ({
+  setOnLoad,
+  setServerError,
+  setPeopleLength,
+}) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [selectedPerson, setSelectedPerson] = useState('');
+
+  const findParrent = (parentName: string) => {
+    return (people.find(person => person.name === parentName))?.slug;
+  };
+
+  useEffect(() => {
+    setOnLoad(true);
+
+    getPeople()
+      .then(value => {
+        setPeople(value);
+        setPeopleLength(value.length);
+      })
+      .catch(() => setServerError(true))
+      .finally(() => setOnLoad(false));
+  }, [people.length]);
```

Comment:
Let’s use `async await` all over the code. It gives you a simpler way to work with asynchronous promise-based code.